server/tserver/src/main/java/org/apache/accumulo/tserver/tablet/ScanDataSource.java

```diff
@@ -234,32 +239,60 @@ private SortedKeyValueIterator<Key,Value> createIterator() throws IOException {
     }
   }
 
-  @Override
-  public void close(boolean sawErrors) {
-
+  private void returnIterators() {
     if (memIters != null) {
-      tablet.returnMemIterators(memIters);
-      memIters = null;
-      tablet.returnFilesForScan(fileReservationId);
-      fileReservationId = -1;
-    }
-
-    synchronized (tablet) {
-      if (tablet.removeScan(this) == 0) {
-        tablet.notifyAll();
+      log.trace("Returning mem iterators for {}, scanId:{}, fid:{}", tablet.getExtent(),
+          scanDataSourceId, fileReservationId);
+      try {
+        tablet.returnMemIterators(memIters);
+      } finally {
+        memIters = null;
+        log.trace("Returning file iterators for {}, scanId:{}, fid:{}", tablet.getExtent(),
+            scanDataSourceId, fileReservationId);
+        try {
+          tablet.returnFilesForScan(fileReservationId);
+        } catch (IllegalArgumentException | IllegalStateException e) {
+          if (log.isTraceEnabled()) {
```

Comment:
Its only logging at warn if trace is enabled

```suggestion
          if (log.isWarnEnabled()) {
```